pulpcore/cli/rpm/repository.py

```diff
@@ -50,6 +57,17 @@
 )
 
 
+def _content_callback(ctx: click.Context, param: click.Parameter, value: Any) -> Any:
+    if value:
+        pulp_ctx = ctx.find_object(PulpContext)
+        assert pulp_ctx is not None
+        ctx.obj = PulpRpmPackageContext(pulp_ctx, entity={"package_href": value})
```

Comment:
This may solve your problem here.

```suggestion
        ctx.obj = PulpRpmPackageContext(pulp_ctx, pulp_href=value})
```

But it's probably not meant that way: This callback was meant to lookup the content by its "natural key" (is it nevra here?), while you created an href-lookup, which is another valid approach currently missing on the file commands.

Reply:
"natural key" for RPMs gets complicated in the wild - you would think it would just 'nevra in repo', but you can have RPMs with the same NEVRA but different checksums, and RPMs with the same NEVRA/checksum but different relative_paths. This is because the "rules" for rpm-repo-layout are...more like a set of **guidelines**, really.

I would eventually like to see a more 'natural' approach available to CLI users - maybe an "add to repo" that takes a JSON file of just path-to-rpm-file and relative-path? - that hides things like sha256 and hrefs and the content-modify call details from the casual CLI user. However, the approach here is def. valid, and not having this API covered is absolutely a Gaping Hole in rpm-support in the CLI - so I'm happy to have it! We can add "higher level" commands at a later date.